Sources/YPersistence/Manager/PersistenceManager+Sync.swift

```diff
@@ -63,7 +66,97 @@ extension PersistenceManager {
         )
         return try fetchRecords(
             predicate: predicate,
-            sortDescriptors: T.sort?.descriptors
+            sortDescriptors: T.sort?.descriptors,
+            context: context
         )
     }
+
+    /// Update records that have been marked for upload since the last sync.
+    ///
+    /// May be safely called from any thread.
+    /// If no context is provided, a worker context will be created.
+    /// - Parameters:
+    ///   - entity: the Core Data entity to be fetched
+    ///   - uids: array of uids to update
+    ///   - context: optional managed context to use. Default is `nil`.
+    /// - Throws: any error executing the fetch or save request
+    public func markRecordsAsUploaded<T: SyncRecord>(
+        entity: T.Type,
+        uids: [T.UidType],
+        context: NSManagedObjectContext? = nil
+    ) throws {
+        let context = context ?? workerContext
+
+        let predicate = NSPredicate(format: "\(T.isUploadedKey) == false && \(entity.uidKey) in %@", uids)
+        
+        let records: [T] = try fetchRecords(
+            predicate: predicate,
+            context: context
+        )
+
+        for record in records {
+            record.isUploaded = true
+        }
+
+        try context.saveChangesAndWait()
+    }
+
+    /// Update records that have been marked for deletion since the last sync.
+    ///
+    /// May be safely called from any thread
+    /// If no context is provided, a worker context will be created.
+    /// - Parameters:
+    ///   - entity: the Core Data entity to be fetched
+    ///   - uids: array of uids to update
+    ///   - context: optional managed context to use. Default is `nil`.
+    /// - Throws: any error executing the fetch or save request
+    public func markRecordsAsDeleted<T: SyncRecord>(
+        entity: T.Type,
+        uids: [T.UidType],
+        context: NSManagedObjectContext? = nil
+    ) throws {
+        let context = context ?? workerContext
+
+        let predicate = NSPredicate(format: " \(T.wasDeletedKey) == false && \(entity.uidKey) in %@", uids)
+
+        let records: [T] = try fetchRecords(
+            predicate: predicate,
+            context: context
+        )
+
+        for record in records {
+            record.wasDeleted = true
+            record.isUploaded = false
+        }
+
+        try context.saveChangesAndWait()
+    }
+
+    /// Update all records as deleted
+    ///
+    /// May be safely called from any thread
```

Comment:
```suggestion
    /// May be safely called from any thread.
```
Missing period (needed to keep the next sentence from being combined with this, in QuickLook, the line breaks are ignored.